mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java

```diff
@@ -205,9 +205,9 @@ public ReIndexerPerformer(MailboxManager mailboxManager,
             .map(message -> new ReIndexingEntry(mailbox, mailboxSession, message));
     }
 
-    private Mono<Task.Result> reIndexMessages(Flux<ReIndexingEntry> entries, RunningOptions runningOptions, ReprocessingContext reprocessingContext) {
-        return throttle(entries, Duration.ofSeconds(1), runningOptions.getMessagesPerSecond())
-            .flatMap(entry -> reIndexMessage(entry.getMailboxSession(), entry.getMailbox(), reprocessingContext, entry.getMessage()))
+    private Mono<Task.Result> reIndexMessages(Flux<ReIndexingEntry> entriesToIndex, RunningOptions runningOptions, ReprocessingContext reprocessingContext) {
+        return throttle(entriesToIndex, Duration.ofSeconds(1), runningOptions.getMessagesPerSecond())
```

Comment:
idem  ReactorUtils::throttle could be used

Reply:
Could be done in a follow up refacto, lets not stack dependencies